webapp/src/main/java/org/apache/atlas/web/rest/DiscoveryREST.java

```diff
@@ -455,6 +456,66 @@ public AtlasSearchResult indexSearch(@Context HttpServletRequest servletRequest,
         }
     }
 
+
+    /**
+     * Index based search for query direct on Elasticsearch Edge index
+     *
+     * @param parameters Index Search parameters @IndexSearchParams.java
+     * @return Atlas search result
+     * @throws AtlasBaseException
+     * @HTTP 200 On successful search
+     */
+    @Path("/relationship/indexsearch")
+    @POST
+    @Timed
+    public AtlasSearchResult relationshipIndexSearch(@Context HttpServletRequest servletRequest, IndexSearchParams parameters) throws AtlasBaseException {
+        AtlasPerfTracer perf = null;
+        long startTime = System.currentTimeMillis();
+
+        RequestContext.get().setIncludeMeanings(!parameters.isExcludeMeanings());
+        RequestContext.get().setIncludeClassifications(!parameters.isExcludeClassifications());
+        RequestContext.get().setIncludeClassificationNames(parameters.isIncludeClassificationNames());
+        try     {
+            if (AtlasPerfTracer.isPerfTraceEnabled(PERF_LOG)) {
+                perf = AtlasPerfTracer.getPerfTracer(PERF_LOG, "DiscoveryREST.relationshipIndexSearch(" + parameters + ")");
+            }
+
+            if (StringUtils.isEmpty(parameters.getQuery())) {
+                throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Invalid search query");
+            }
+
+            if(LOG.isDebugEnabled()){
+                LOG.debug("Performing relationship indexsearch for the params ({})", parameters);
+            }
+            return discoveryService.directRelationshipIndexSearch(parameters);
+
+        } catch (AtlasBaseException abe) {
+            throw abe;
+        } catch (Exception e) {
+            AtlasBaseException abe = new AtlasBaseException(e.getMessage(), e.getCause());
+            throw abe;
+        } finally {
+            if(CollectionUtils.isNotEmpty(parameters.getUtmTags())) {
+                AtlasPerfMetrics.Metric indexsearchMetric = new AtlasPerfMetrics.Metric(RELATIONSHIP_INDEXSEARCH_TAG_NAME);
+                indexsearchMetric.addTag("utmTag", "other");
+                indexsearchMetric.addTag("source", "other");
+                for (String utmTag : parameters.getUtmTags()) {
+                    if (TRACKING_UTM_TAGS.contains(utmTag)) {
+                        indexsearchMetric.addTag("utmTag", utmTag);
+                        break;
+                    }
+                }
+                if (parameters.getUtmTags().contains(UTM_TAG_FROM_PRODUCT)) {
+                    indexsearchMetric.addTag("source", UTM_TAG_FROM_PRODUCT);
+                }
+                indexsearchMetric.addTag("name", RELATIONSHIP_INDEXSEARCH_TAG_NAME);
+                indexsearchMetric.setTotalTimeMSecs(System.currentTimeMillis() - startTime);
```

Comment:
Check this with Suman

Reply:
I think we can skip it for now, we have added this metric to track indexsearch only which is very important. Do you think this endpoint would be very important to track by utmTags?